src/Chapter3.hs

```diff
@@ -908,8 +996,17 @@ Implement instances of "Append" for the following types:
 
 -}
 class Append a where
-    append :: a -> a -> a
+  append :: a -> a -> a
+
+newtype Gold = Gold Int
+
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append (Gold x) (Gold y) = Gold (x + y)
 
+instance Append [a] where
+  append :: [a] -> [a] -> [a]
+  append [x] [y] = x : [y]
```

Comment:
This solution doesn't cover all possible cases. It will fail on empty lists and on lists with more than one element inside.
You probably want to check out `++` operator 😉